static/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -353,12 +353,19 @@ class WidgetQueries extends React.Component<Props, State> {
             return prevState;
           }
 
-          const timeseriesResults = (prevState.timeseriesResults ?? []).concat(
-            transformResult(widget.queries[i], rawResults)
+          const timeseriesResults = [...(prevState.timeseriesResults ?? [])];
+          const transformedResult = transformResult(
+            widget.queries[requestIndex],
+            rawResults
           );
+          // We insert at a specific index instead of just pushing to timeseriesResults since we need to preserve the order of results because order determines color
+          transformedResult.forEach((result, resultIndex) => {
+            timeseriesResults[requestIndex * transformedResult.length + resultIndex] =
+              result;
+          });
```

Comment:
Won't transformedResult will always be an array of 1 element? Since we are only transforming the result of one widget query? So is this equivalent to something like:

```suggestion
          timeseriesResults[requestIndex] = result[0];
```

Reply:
jk, literally had a dream about this PR last night -- we need something like what you have to account for multiple y-axes

Reply:
Can we just add a more descriptive comment because it's not totally obvious what's happening

Reply:
Yup, updated the comment!